### PR TITLE
refactor(facade): cleanup Intl facade

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

**Comment by vicb** [comment]
@pkozlowski-opensource could you please take a look. No functional change - will probably need more work.

File: modules/@angular/facade/src/intl.ts [hunk 1]
```diff
@@ -37,10 +37,13 @@ export class NumberFormatter {
     return new Intl.NumberFormat(locale, options).format(num);
   }
 }
-var DATE_FORMATS_SPLIT =
+
+type DateFormatterFn = (date: Date, locale: string) => string;
+
+const DATE_FORMATS_SPLIT =
```
Comment (DzmitryShylovich) [inline modules/@angular/facade/src/intl.ts]: Sometimes date pipe test fails with the following error http://prnt.sc/cu6pfs
That's the only regexp used in date pipe.
Probably related to https://github.com/andyearnshaw/Intl.js/issues/231

Comment (vicb) [inline modules/@angular/facade/src/intl.ts]: Thanks for the tip, I was actually chasing this error !

Comment (vicb) [inline modules/@angular/facade/src/intl.ts]: @DzmitryShylovich Could you please create a separate issue on GH for the pb you mention ? Thanks

**Comment by vsavkin** [comment]
Merged https://github.com/angular/angular/commit/7b8dae19af54b93e1120b8203a1f4494d2d42294

**Pull request closed without merging**